library/core/src/iter/traits/collect.rs

```diff
@@ -416,17 +421,32 @@ where
     /// assert_eq!(b, [2, 5, 8]);
     /// assert_eq!(c, [3, 6, 9]);
     /// ```
-    fn extend<T: IntoIterator<Item = (A, B)>>(&mut self, into_iter: T) {
+    fn extend<T: ~const IntoIterator<Item = (A, B)>>(&mut self, into_iter: T)
+    where
+        T::IntoIter: ~const Destruct,
+    {
         let (a, b) = self;
         let iter = into_iter.into_iter();
 
-        fn extend<'a, A, B>(
-            a: &'a mut impl Extend<A>,
-            b: &'a mut impl Extend<B>,
-        ) -> impl FnMut((), (A, B)) + 'a {
-            move |(), (t, u)| {
-                a.extend_one(t);
-                b.extend_one(u);
+        pub struct ExtendFold<'a, EA, EB>(&'a mut EA, &'a mut EB);
```

Comment:
Is this a temporary thing to be removed later? If so it could use a FIXME(#issuenumber) for the feature needed to solve it.

If this isn't meant to be temporary I have additional concerns.

Reply:
Yes, this is temporary. This could be improved and made to look better using `ConstFnMutClosure`, but in the long run we plan to replace these with const closures.

Reply:
I've improved it.